nucypher/cli/commands/dkg.py

```diff
@@ -0,0 +1,343 @@
+
+GlobalLoggerSettings.start_console_logging()
+emitter = StdoutEmitter(verbosity=2)
+
+
+option_eth_staking_network = click.option(
+    "--eth-staking-network",
+    "eth_staking_network",
+    help="ETH staking network",
+    type=click.Choice(["tapir", "lynx"]),
+    default="lynx",
+)
+
+option_coordinator_provider = click.option(
+    "--coordinator-provider",
+    "coordinator_provider_uri",
+    help="Coordinator network provider URI",
+    type=click.STRING,
+    required=True,
+)
+
+option_coordinator_network = click.option(
+    "--coordinator-network",
+    "coordinator_network",
+    help="Coordinator network",
+    type=click.Choice(["mumbai"]),
+    default="mumbai",
+)
+
+
+def get_agency(
+    coordinator_network, coordinator_provider_uri, eth_staking_network, eth_provider_uri
+):
+    coordinator_agent = ContractAgency.get_agent(
+        agent_class=CoordinatorAgent,
+        registry=InMemoryContractRegistry.from_latest_publication(
+            network=coordinator_network
+        ),
+        provider_uri=coordinator_provider_uri,
+    )  # type: CoordinatorAgent
+
+    staking_network_registry = InMemoryContractRegistry.from_latest_publication(
+        network=eth_staking_network
+    )
+    application_agent = ContractAgency.get_agent(
+        agent_class=PREApplicationAgent,
+        registry=staking_network_registry,
+        provider_uri=eth_provider_uri,
+    )  # type: PREApplicationAgent
+
+    return coordinator_agent, application_agent, staking_network_registry
+
+
+@click.group()
+def dkg():
+    """DKG management commands"""
+
+
+@dkg.command()
+@option_eth_provider_uri(required=True)
+@option_eth_staking_network
+@option_coordinator_provider
+@option_coordinator_network
+@click.option(
+    "--signer",
+    "signer_uri",
+    "-S",
+    help="Signer URI for initiating a new ritual with Coordinator contract",
+    default=None,
+    type=click.STRING,
+)
+@click.option(
+    "--dkg-size",
+    "dkg_size",
+    "-n",
+    help="Number of nodes to participate in ritual",
+    type=click.INT,
+    default=0,
+)
+@click.option(
+    "--num-rituals",
+    "num_rituals",
+    "-r",
+    help="The number of rituals to initiate",
+    type=click.INT,
+    default=1,
+)
+def init(
+    eth_provider_uri,
+    eth_staking_network,
+    coordinator_provider_uri,
+    coordinator_network,
+    signer_uri,
+    dkg_size,
+    num_rituals,
+):
+    # if creating ritual(s)
+    if signer_uri is None:
+        raise click.BadOptionUsage(
+            option_name="--ritual-id, --signer",
+            message=click.style(
+                "--signer must be provided to create new ritual when --ritual-id is not provided",
+                fg="red",
+            ),
+        )
+    if dkg_size <= 1 or dkg_size % 2 != 0:
+        raise click.BadOptionUsage(
+            option_name="--dkg-size",
+            message=click.style("DKG size must be > 1 and a power of 2", fg="red"),
```

Comment:
The power-of-2 restriction was removed in https://github.com/nucypher/ferveo/pull/134. Unsure about its current support in nucypher, but I think it'd be better to remove the restriction here.

Reply:
✔️